crates/cast/src/helpers/build.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{Context, Result};
+use scarb_artifacts::{get_contracts_map, target_dir_for_package, StarknetContractArtifacts};
+use scarb_metadata::{Metadata, PackageMetadata};
+use scarb_ui::args::PackagesFilter;
+use std::collections::HashMap;
+use std::fs;
+use std::process::{Command, Stdio};
+
+use super::constants::LIB_CONTRACT_ARTIFACTS_NAME;
+
+#[allow(clippy::too_many_lines)]
+pub fn build(
+    metadata: &Metadata,
+    package: &PackageMetadata,
+    for_scripts: bool,
+) -> Result<HashMap<String, StarknetContractArtifacts>> {
+    let filter = PackagesFilter::generate_for::<Metadata>([package].into_iter());
+
+    let command_result = Command::new("scarb")
+        .env("SCARB_PACKAGES_FILTER", filter.to_env())
+        .arg("--manifest-path")
+        .arg(&package.manifest_path)
+        .arg("build")
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .output()
+        .context("Failed to start building contracts with Scarb")?;
+
+    let result_code = command_result
+        .status
+        .code()
+        .context("Failed to obtain status code from scarb build")?;
+    let result_msg = String::from_utf8(command_result.stdout)?;
+    if result_code != 0 {
+        anyhow::bail!(
+            "Scarb build returned non-zero exit code: {} - error message: {}",
+            result_code,
+            result_msg
+        );
+    }
+
+    let mut contracts = get_contracts_map(metadata, &package.id)?;
+
+    if for_scripts {
+        // Insert lib "contract"
```

Comment:
I think it's pretty clear from the code what's going on here. We may consider extracting this to a separate function to improve readability